backend/app/api/src/helpers/AdminPermissionChecker.ts

```diff
@@ -187,26 +187,130 @@ export class AdminPermissionChecker {
         return false;
     }
 
-    async canAccessEvent(event: Event, permissionLevel: PermissionLevel = PermissionLevel.Read): Promise<boolean> {
-        // Check permissions aren't scoped to a specific organization, and they mismatch
-        if (!this.checkScope(event.organizationId)) {
-            return false
-        }
+    /**
+     * Will throw error if not allowed to edit/add/delete this event
+     * @param event 
+     * @returns Organization if event for specific organization, else null
+     * @throws error if not allowed to write this event
+     */
+    async tryAdminEvent(event: Event, options: {organization?: Organization} = {}): Promise<Organization | null> {
```

Comment:
Ik vind de optie organization niet zo duidelijk - het lijkt een soort cache te zijn, maar ik weet niet of dat zo een goed idee is